pkg/cloud/services/eks/tags.go

```diff
@@ -73,6 +76,35 @@ func (s *NodegroupService) reconcileTags(ng *eks.Nodegroup) error {
 	return updateTags(s.EKSClient, ng.NodegroupArn, aws.StringValueMap(ng.Tags), tags)
 }
 
+func (s *NodegroupService) reconcileAsgTags(ng *eks.Nodegroup) error {
+	if s.scope.ManagedMachinePool.Spec.AdditionalAsgTags == nil {
+		return nil
+	}
+	trueVal := true
+	asgType := "auto-scaling-group"
+	for _, asg := range ng.Resources.AutoScalingGroups {
+		input := &autoscaling.CreateOrUpdateTagsInput{}
+		for k, v := range s.scope.ManagedMachinePool.Spec.AdditionalAsgTags {
+			// The k/vCopy is used to address the "Implicit memory aliasing in for loop" issue
+			// https://stackoverflow.com/questions/62446118/implicit-memory-aliasing-in-for-loop
+			kCopy := k
+			vCopy := v
+			input.Tags = append(input.Tags, &autoscaling.Tag{
+				Key:               &kCopy,
+				PropagateAtLaunch: &trueVal,
+				ResourceId:        asg.Name,
+				ResourceType:      &asgType,
```

Comment:
similarly you can do for `asgtype` using func `pointer.String("foo")` from lib `"k8s.io/utils/pointer"`

Reply:
done